### PR TITLE
feat(core): ldml C++ validate list / key2  🙀

### PR DESCRIPTION
**Pull request by srl295**
- need helper objs still

For: #7985

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by rc-swag** [comment]
Late to the party here but I am not following what `key2` is from the spec.

**Comment by srl295** [comment]
“keys” is the keybag minimized / optimized for keyman. “key2” is a
different structure that contains all of the attributes. For example ids
and such.

File: core/src/kmx/kmx_plus.cpp [hunk 1]
```diff
@@ -575,15 +575,27 @@ COMP_KMXPLUS_LAYR_Helper::getKey(KMX_DWORD key) const {
...
+  // TODO-LDML: further validation in the COMP_KMXPLUS_KEY2_Helper helper obj
   return true;
 }
 
 bool
 COMP_KMXPLUS_LIST::valid(KMX_DWORD _kmn_unused(length)) const {
-  // TODO-LDML more to do here
-  DebugLog("TODO-LDML: list");
+  if (header.size <
+      sizeof(*this) + (sizeof(COMP_KMXPLUS_LIST_ITEM) * listCount) + (sizeof(COMP_KMXPLUS_LIST_INDEX) * indexCount)) {
```
Comment (mcdurdin) [inline core/src/kmx/kmx_plus.cpp:593]: Can you make this test match the style of lines 578-582?

**Review by mcdurdin — APPROVED** [review]

File: core/src/kmx/kmx_plus.cpp (hunk 1 above)
Comment (srl295) [reply core/src/kmx/kmx_plus.cpp]: done

**Pull request merged**